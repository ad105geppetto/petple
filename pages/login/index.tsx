import Signup from "../../src/components/units/signup/Signup.container";

export default function LoginPage() {
  return <Signup isSignup={false} />;
}
