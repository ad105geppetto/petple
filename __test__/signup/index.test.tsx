import "@testing-library/jest-dom";
import { MockedProvider } from "@apollo/client/testing";
import { fireEvent, render, screen, waitFor } from "@testing-library/react";
import SignupUI from "../../src/components/units/signup/Signup.presenter";
import Signup from "../../src/components/units/signup/Signup.container";
import { mocks, propsMock } from "./login.mocks";
import { useRouter } from "next/router";
import { RecoilRoot } from "recoil";

jest.mock("next/router", () => ({
  useRouter: jest.fn(),
}));

describe("로그인 Unit Test & Integration Test", () => {
  it("회원가입 스냅샷 테스트", () => {
    const result = render(
      <RecoilRoot>
        <MockedProvider mocks={mocks} addTypename={false}>
          <SignupUI {...propsMock} />
        </MockedProvider>
      </RecoilRoot>
    );

    expect(result.container).toMatchSnapshot();
  });

  it("필수 정보 입력 후 가입하기 버튼 클릭시, 페이지 이동합니다.", async () => {
    const router = {
      replace: jest.fn(),
    };
    (useRouter as jest.Mock).mockReturnValue(router);

    render(
      <RecoilRoot>
        <MockedProvider mocks={mocks} addTypename={false}>
          <Signup isSignup={true} />
        </MockedProvider>
      </RecoilRoot>
    );

    const email = screen.getByPlaceholderText("이메일");
    fireEvent.blur(email, { target: { value: "test@test.com" } });
    const password = screen.getByPlaceholderText("비밀번호");
    fireEvent.blur(password, { target: { value: "1234" } });
    const passwordCheck = screen.getByPlaceholderText("비밀번호 확인");
    fireEvent.blur(passwordCheck, { target: { value: "1234" } });
    const name = screen.getByPlaceholderText("이름");
    fireEvent.blur(name, { target: { value: "lee" } });

    const onClickLogin = screen.getByText("가입하기");
    fireEvent.click(onClickLogin);

    await waitFor(() => {
      expect(router.replace).toHaveBeenCalledWith("/boards");
    });
  });
});
