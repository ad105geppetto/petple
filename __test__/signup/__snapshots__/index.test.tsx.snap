// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`로그인 Unit Test & Integration Test 회원가입 스냅샷 테스트 1`] = `
<div>
  <div
    class="css-g4fya0"
  >
    
    <h2
      class="css-lqiix7"
    >
      펫플 
      회원가입
    </h2>
    <div
      class="css-1dy1q9i"
    >
      <h3
        class="css-1lup52f"
      >
        <label
          class="css-1j2v978"
          for="email"
        >
          이메일
          *
        </label>
      </h3>
      <input
        autocomplete="off"
        class="css-10uky0u"
        id="email"
        placeholder="이메일"
        type="text"
      />
      <h3
        class="css-1lup52f"
      >
        <label
          class="css-1j2v978"
          for="password"
        >
          비밀번호
          *
        </label>
      </h3>
      <input
        autocomplete="off"
        class="css-10uky0u"
        id="password"
        placeholder="비밀번호"
        type="password"
      />
      <h3
        class="css-1lup52f"
      >
        <label
          class="css-1j2v978"
          for="password_check"
        >
          비밀번호 확인*
        </label>
      </h3>
      <input
        autocomplete="off"
        class="css-10uky0u"
        id="password_check"
        placeholder="비밀번호 확인"
        type="password"
      />
      <h3
        class="css-1lup52f"
      >
        <label
          class="css-1j2v978"
          for="name"
        >
          이름*
        </label>
      </h3>
      <input
        autocomplete="off"
        class="css-10uky0u"
        id="name"
        placeholder="이름"
        type="text"
      />
    </div>
    <div
      class="css-1dy1q9i"
    >
      <button
        class="css-1u35s9i"
      >
        가입하기
      </button>
    </div>
  </div>
</div>
`;
